### PR TITLE
Add P2 GitHub connection

### PR DESCRIPTION
**Pull request by unDemian**
#### Changes proposed in this Pull Request

Markup on 2021-11-10 at 3:02:49 PM
Add support for Github connection for P2 hubs.

#### Testing instructions
- Build and push 4015-gh-Automattic/p2
- Navigate to `/marketing/connections/<site-url>` for a P2 hub - there should be a Github connection **(should be available only for automatticians)**
- Navigate to `/marketing/connections/<site-url>` for a P2 site - there should be the regular 3rd party connections (facebook/twitter)
- Navigate to `/marketing/connections/<site-url>` for a regular site - there should be the regular 3rd party connections (facebook/twitter) and make sure they work

For P2 hubs, try to connect/disconnect a Github account.

### CODE REVIEW

File: client/my-sites/marketing/connections/service-description.jsx [hunk 1]
```diff
@@ -146,6 +146,17 @@ class SharingServiceDescription extends Component {
 					comment: 'Get slack notifications on new P2 posts.',
 				} );
 			},
+			p2_github: function () {
+				if ( this.props.numberOfConnections > 0 ) {
+					return this.props.translate( 'Workspace connected to Github.', {
```
Comment (a8ci18n) [inline client/my-sites/marketing/connections/service-description.jsx:151]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Workspace connected to Slack.' )` *ES Score: 7*

File: client/my-sites/marketing/connections/service-description.jsx [hunk 2]
```diff
@@ -146,6 +146,17 @@ class SharingServiceDescription extends Component {
...
+						comment: 'Embed Github Issues in P2 posts.',
+					} );
+				}
+
+				return this.props.translate( 'Connect this workspace to your Github.', {
```
Comment (a8ci18n) [inline client/my-sites/marketing/connections/service-description.jsx:156]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Connect this workspace to your Slack.' )` *ES Score: 7*

**Comment by nefeline** [comment]
## Test Results:

> Navigate to /marketing/connections/<site-url> for a P2 hub - there should be a Github connection (should be available only for automatticians)

🗒️  Navigated to a P2 hub. Confirmed the GitHub connection is displayed as expected, although the logo is missing:

Screenshot 2021-12-13 at 13 00 28

✅  Confirmed the GitHub connection is not available when connecting with a non-automattician account:

Screenshot 2021-12-13 at 13 01 12

> Navigate to /marketing/connections/<site-url> for a P2 site - there should be the regular 3rd party connections (facebook/twitter)

✅  Navigated to a P2 site, confirmed all regular 3rd party connections are displayed.

Screenshot 2021-12-13 at 13 02 08

> Navigate to /marketing/connections/<site-url> for a regular site - there should be the regular 3rd party connections (facebook/twitter) and make sure they work

✅ Confirmed Facebook and Twitter connections are working as expected on a regular site.

> For P2 hubs, try to connect/disconnect a Github account.

🗒️ When the arrow on the right is clicked, nothing happens:

Screenshot 2021-12-13 at 13 03 35

🗒️ When the "Connect" button is clicked, a new window is opened.

❌ When following the normal flow for enabling the GitHub connection, an error is thrown: "The provided keyring token is not for a Publicize service.".

Screenshot 2021-12-13 at 13 04 28

**Comment by unDemian** [comment]
> 🗒️ Navigated to a P2 hub. Confirmed the GitHub connection is displayed as expected, although the logo is missing:
> 🗒️ When the arrow on the right is clicked, nothing happens:

Interesting, it seems like something was not build correctly, not sure why, maybe try a `yarn install` before `yarn start`. It seems to be loading the image in the testing environment https://calypso.live?image=registry.a8c.com/calypso/app:build-21806

> ❌ When following the normal flow for enabling the GitHub connection, an error is thrown: "The provided keyring token is not for a Publicize service.".

Hmm.. I could not replicate this, have you build and pushed 4015-gh-Automattic/p2/files and sandbox the public-api?

**Review by nefeline — APPROVED** [review]
> Interesting, it seems like something was not build correctly, not sure why, maybe try a yarn install before yarn start. It seems to be loading the image in the testing environment https://calypso.live?image=registry.a8c.com/calypso/app:build-21806

Yeah, something was off with the build indeed: running `yarn install` did the trick ✨ 

> Navigate to /marketing/connections/<site-url> for a P2 hub - there should be a Github connection (should be available only for automatticians)

✅ Confirmed the GitHub connection is visible and works as expected. This option is not available to non-automatticians.

Screenshot 2021-12-16 at 15 30 25

Note: only the first two letters of my username are displayed on the "connected by" message.

> Navigate to /marketing/connections/<site-url> for a P2 site - there should be the regular 3rd party connections (facebook/twitter)

✅ Confirmed that all regular 3dr party connections are available for a single P2 site.

> Navigate to /marketing/connections/<site-url> for a regular site - there should be the regular 3rd party connections (facebook/twitter) and make sure they work

✅ Confirmed connections are working properly on regular sites as well. Connections with Facebook and Twitter were tested and are fully functional.

> For P2 hubs, try to connect/disconnect a Github account.

I was able to successfully connect and disconnect from GitHub on P2 hubs.

Solid work! 👏

**Comment by unDemian** [comment]
Thanks for taking the time.

> Note: only the first two letters of my username are displayed on the "connected by" message.

Should be fixed by [D72025](D72025-code)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7053012

Thank you @unDemian for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7053012).

**Pull request merged**